File: app/src/main/java/me/ameriod/lib/mvp/app/person/PersonInteractor.java
package me.ameriod.lib.mvp.app.person;

import android.support.annotation.NonNull;

import io.reactivex.Observable;
import me.ameriod.lib.mvp.app.api.SwapiClient;
import me.ameriod.lib.mvp.app.models.Person;

public class PersonInteractor implements PersonContract.Interactor {
    @NonNull
    private SwapiClient client;
    @NonNull
    private String id;

    public PersonInteractor(@NonNull SwapiClient client, @NonNull String id) {
        this.client = client;
        this.id = id;
    }

    public PersonInteractor(@NonNull String id) {
        this(new SwapiClient(), id);
    }

    @Override
    public Observable<Person> getPerson() {
        return client.getPerson(id);
    }
}

File: app/src/main/java/me/ameriod/lib/mvp/app/person/PersonInteractor.kt
package me.ameriod.lib.mvp.app.person

import io.reactivex.Observable
import me.ameriod.lib.mvp.app.api.SwapiClient
import me.ameriod.lib.mvp.app.models.Person

class PersonInteractor(private val client: SwapiClient, private val id: String) : PersonContract.Interactor {

    constructor(id: String) : this(SwapiClient(), id)

    override fun getPerson(): Observable<Person> {
        return client.getPerson(id)
    }
}